light-client/src/operations/voting_power.rs

```diff
@@ -1,19 +1,75 @@
 use crate::{
     bail,
     predicates::errors::VerificationError,
-    types::{SignedHeader, ValidatorSet},
+    types::{Commit, SignedHeader, TrustThreshold, ValidatorSet},
 };
 
-use anomaly::BoxError;
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+use std::fmt;
+
+use tendermint::block::CommitSig;
+use tendermint::lite::types::TrustThreshold as _;
 use tendermint::lite::types::ValidatorSet as _;
+use tendermint::vote::{SignedVote, Vote};
+
+#[derive(Copy, Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct VotingPower {
+    pub total: u64,
+    pub tallied: u64,
+    pub trust_threshold: TrustThreshold,
+}
+
+impl fmt::Display for VotingPower {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "VotingPower(total={} tallied={} trust_threshold={})",
+            self.total, self.tallied, self.trust_threshold
+        )
+    }
+}
 
 pub trait VotingPowerCalculator: Send {
     fn total_power_of(&self, validators: &ValidatorSet) -> u64;
-    fn voting_power_in(
+
+    fn check_enough_trust(
```

Comment:
The name and the result of this function were for me a bit misleading as the name would suggest a boolean value (or in this case `Ok()` or `Error(...)`) and it is more or less used as such since the result value in the Ok case is just ignored.